### DISCUSSION TITLE
MINTEMP ERR: nozzle temperature too low

### DISCUSSION THREAD

**Discussion by Apie56** [mriscoc/Ender3V2S1#1092] [Q&A]
Good day.

Please find attached my configuration files. I continue to get a MINTEMP error on my Ender 3 v2.

Sometimes, when I turn off the printer and turn it on again, I no longer get the error, but also sometimes, when I use the reboot function on the screen, I no longer get the error. However, most of the time, the printer continues to give me the error. The strange thing is, that the PID works perfectly fine. I have run multiple cycles from 195 degrees C up to 275 degrees C, fan on and off, bed heated and not, with no issues.

As soon as I start heating the hotend; manually, through a preset, or print start; I get the MINTEMP error.

The thermistor is working, and I have tried with a new thermistor and still get the same resulting error. You will note that I have enabled high thermistor support in the attached configuration_adv.h file to try and counteract this, but it does not make a difference.

I have also tried [this solution](https://github.com/mriscoc/Ender3V2S1/discussions/1022), I've tried flashing firmware and compared my configurations to precompiled configs, with no success. The firmware I have flashed is from the September release.

Additional:
Board: creality v4.2.2 RE
Hotend: creality sprite
Thermistor: 100k ntc
Heater: 50W
Silicone sock: yes
Enclosure: yes

Thanks in advance.
[Configuration.zip](https://github.com/mriscoc/Ender3V2S1/files/12618095/Configuration.zip)

**Comment by tokolist** [comment]
It seems to be Marlin temperature reading instability issue. On 20230803 (and earlier) I was facing it once per a couple of month. On 20230925 I was getting it constantly until I set below value in `Configuration_adv.h` and it got fixed for me. I also added it to my `_printers` configuration to make it permanent solution. You might want to test lower values if you care about safety more than me :)

```C++
#define MAX_CONSECUTIVE_LOW_TEMPERATURE_ERROR_ALLOWED 10
```

```JavaScript
  {
    "op": "CustomVal",
    "searchfor": "MAX_CONSECUTIVE_LOW_TEMPERATURE_ERROR_ALLOWED",
    "value": "10",
    "comment": "Fixes MINTEMP issue caused by unstable temperature readings"
  }
```

**Reply by tokolist** [reply]
Unfortunately it was working quite well only till I rebooted printer. I also gave `PREHEAT_TIME_HOTEND_MS 30000` a try on top of that but no luck :( I rolled firmware back to 20230803 for the time being. I will get back to it when have some time, but so far that looks like Marlin core issue.

**Reply by Apie56** [reply]
Thanks for your reply.

This seemed to work for 2 prints but thereafter I started getting the error again. I had to roll back my firmware too. and all is working well now. Hopefully, its something that can be fixed overall rather than just being an issue on my side.

**Reply by tokolist** [reply]
Precompiled `Ender3V2-422-BLTUBL-MPC-20230904.bin` seems to be working stable and `M104 S150` handles correctly even after few reboots. Precompiled one doesn't work for me since I need to make configurational changes for my setup. I will do my best to figure out what is the difference between precompiled one and source code.

**Reply by tokolist** [reply]
That might sound weird but I double confirmed that. Issue appears only when `AUTO_BED_LEVELING_BILINEAR` is used instead of `AUTO_BED_LEVELING_UBL`. Literally when I change from

```c++
#define AUTO_BED_LEVELING_BILINEAR
//#define AUTO_BED_LEVELING_UBL
```

to

```c++
//#define AUTO_BED_LEVELING_BILINEAR
#define AUTO_BED_LEVELING_UBL
```

it works as expected.

@Apie56 are you using UBL or BILINEAR?

**Reply by tokolist** [reply]
Found your configuration attached and you use BILINEAR, please try to change to UBL and let me know if M104 works fine after that.

**Reply by Apie56** [reply]
Yes, I am using Bilinear. I will try UBL asap. I will only be back at the workshop on Friday. 

Is there any specific hardware requirement for UBL?

**Reply by tokolist** [reply]
No specific hardware requirements, but you have to change your gcode. UBL is just more advanced than Bilinear mesh leveling system. As of now I stuck with very simple implementation. I was using Bilinear to get new mesh before each print in startup gcode, so now using UBL in identical way.  I might use advanced UBL features, i.e. store meshes for different configurations and then tilt them on startup to save time as people typically do with UBL.

I made these two changes in order to get it working as before:

1. Selected UBL in Configurator (besides AUTO_BED_LEVELING_UBL it changes few more settings)
2. Replaced G29 with below gcode in startup script
```
G29 P1        ; Do automated probing of the bed.
G29 P3        ; Smart Fill Repeat until all mesh points are filled in, Used to fill unreachable points.
G29 A         ; Activate the UBL System.
```

UBL docs:
https://marlinfw.org/docs/features/unified_bed_leveling.html
https://marlinfw.org/docs/gcode/G029-ubl.html

**Reply by Apie56** [reply]
Thank you tokolist. All is well and working. I just want to clarify: based on other resources, in my start, I should have:

G29 L0; Load UBL mesh values from slot 0
G29 J; Probe 3 points and tilt the mesh to the plane.

is there any reason why this is different from your suggested script?

**Reply by tokolist** [reply]
It's just my preference. In my case mesh is not stored. It's regenerated every time before print starts, so no need to load and tilt it.

For instance, you might want to generate huge 10x10 mesh which you can store in one of six slots. You can do it through gcode or through printer menu. Then you load it in startup script before each print and tilt to compensate potential bed tilting. This will save you a lot of time since regenerating 100 points mesh takes a while.

I'm literally skipping irrelevant steps:

1. I don't need to store and load mesh, because I regenerate it every print.
2. I don't need to tilt it because it's fresh mesh and bed cannot get tilted in such short time. Additionally, I use silicone standoffs, so it's stable for long period of time thus I might not even need to tilt it in long term perspective, but I would do it anyways just in case.

UBL gives you tons of features and it's up to you how to use them :)

**Reply by Apie56** [reply]
thanks so much for your help and insight. Much appreciated.

**Reply by tokolist** [reply]
you are welcome, happy to help

**Reply by whodunyeet** [reply]
Hi, I am facing exactly the same issue with my Ender 3 V2
I've replaced the thermistors, but unable to resolve this. The temperature keeps fluctuating and drops at random instances.
I am not well versed to coding or editing G-codes. Can you please help me resolve this.
Thanks.

**Reply by mistergates** [reply]
> Hi, I am facing exactly the same issue with my Ender 3 V2 I've replaced the thermistors, but unable to resolve this. The temperature keeps fluctuating and drops at random instances. I am not well versed to coding or editing G-codes. Can you please help me resolve this. Thanks.

Also having the same issue on my Ender 3 v2. Temperatures are stable until the print starts, then it rapidly drops and reports nozzle temp too low error.

**Reply by Apie56** [reply]
Reverting to a previous release fixed my issue. Not the best fix, but it works. What firmware/build do you have? Do you have a heat sock? have you checked your thermistor wiring?

**Reply by mistergates** [reply]
@Apie56 which release did you revert to? I was on the July 2024 release, precompiled release build `Ender_422_BLT-ProUI-EX-07-26.bin`. Heat sock is on and thermal wiring is good, I have no issues with creality or jyers firmware.

**Reply by Apie56** [reply]
I reverted to December 2023. I recommend running pid tuning, and if that doesn't work, roll back to 2023. Another thing you can try is increasing the high-temperature support time in configuration_adv.h. It isn't recommended to do so, but could you try to see if it fixes your issue? The following lines are applicable:

 * The solution: Once the temperature reaches the target, start observing.
 * If the temperature stays too far below the target (hysteresis) for too
 * long (period), the firmware will halt the machine as a safety precaution.
 *
 * If you get false positives for "Thermal Runaway", increase
 * THERMAL_PROTECTION_HYSTERESIS and/or THERMAL_PROTECTION_PERIOD
 */
#if ENABLED(THERMAL_PROTECTION_HOTENDS)
  #define THERMAL_PROTECTION_PERIOD 40        // (seconds)
  #define THERMAL_PROTECTION_HYSTERESIS 4     // (°C)
  
  * If you get false positives for "Heating failed", increase WATCH_TEMP_PERIOD
   * and/or decrease WATCH_TEMP_INCREASE. WATCH_TEMP_INCREASE should not be set
   * below 2.
   */
  #define WATCH_TEMP_PERIOD  40               // (seconds)  // Ender Configs
  #define WATCH_TEMP_INCREASE 2               // (°C)
  
  // The number of consecutive low temperature errors that can occur
// before a MINTEMP error is triggered. (Shouldn't be more than 10.)
//#define MAX_CONSECUTIVE_LOW_TEMPERATURE_ERROR_ALLOWED 0

**Reply by Apie56** [reply]
See also the following:
[(https://github.com/mriscoc/Ender3V2S1/discussions/1092#discussioncomment-7110384)](https://github.com/mriscoc/Ender3V2S1/discussions/1092#discussioncomment-7110384)
[(https://github.com/mriscoc/Ender3V2S1/discussions/1092#discussioncomment-7104127)](https://github.com/mriscoc/Ender3V2S1/discussions/1092#discussioncomment-7104127)

**Comment by mriscoc** [comment]
ALM (https://github.com/mriscoc/Ender3V2S1/wiki/Adaptive-Mesh-Leveling-(AML)) could give you better results

[![](https://img.youtube.com/vi/YZ8zBnTRzSQ/hqdefault.jpg)](https://www.youtube.com/watch?v=YZ8zBnTRzSQ)

**Reply by Apie56** [reply]
This is very cool! I will definitely consider it. At the moment, most, if not all my projects utilize the entire bed. But I might just use it for my larger printers.

**Reply by tokolist** [reply]
purchased pack, will be testing it

**Reply by tokolist** [reply]
@mriscoc I have adapted post processing script to PrusaSlicer startup gcode macro, so one doesn't have to install python, libs and configure it. Since it works natively, it also visualizes purge line and it's easy to update right in PrusaSlicer. I think I need your permission to share it publicaly or let me know if I can share it with you privately. You might consider adding it to pack.

Overall I'm happy with results, but I'm still testing it and can see some potential improvements. For instance, purge line for big model is going to be huge and it should be limited by some value.

![image](https://github.com/mriscoc/Ender3V2S1/assets/1922990/3b895910-4840-4dab-b7a7-2b444d6e18de)
![image](https://github.com/mriscoc/Ender3V2S1/assets/1922990/f81385e7-c327-4065-b7fd-78969edc7686)

**Reply by mriscoc** [reply]
Hi, thanks, I the use a script for easy maintain and for keep the start script clean. Python and the libraries are also required for the thumbnail preview. You can share your start script but not here or in ProF social media to avoid confusions.

**Reply by tokolist** [reply]
thanks, I will share it as a secret gist and mention that it's unsupported by Professional Firmware author in case somebody wants to use it

**Reply by tokolist** [reply]
and I will push to purchase AML pack first :)

**Comment by mistergates** [comment]
Thanks for the info! I actually went to the latest release, UBL
pre-compiled version and I'm no longer seeing the issue.
